keyboards/dztech/dz65rgb/keymaps/yuannan/keymap.c

```diff
@@ -0,0 +1,119 @@
+		KC_LSHIFT,		KC_Z,		KC_X,	KC_C,	KC_V,	KC_B,	KC_N,	KC_M,	KC_COMMA,	KC_DOT,	KC_SLASH,	KC_RSHIFT,						KC_UP,		KC_END,
+		KC_CAPSLOCK,	KC_LGUI,	FN_MO13,				KC_SPACE,				FN_MO23,	KC_RALT,	KC_RCTRL,	KC_LEFT,					KC_DOWN,	KC_RIGHT
+    ),
+    [_VIM_EDITING_LAYER] = LAYOUT_65_ansi(
+		KC_GRAVE,	KC_F1,		KC_F2,		KC_F3,		KC_F4,		KC_F5,		KC_F6,		KC_F7,		KC_F8,		KC_F9,		KC_F10,		KC_F11,		KC_F12,		KC_DELETE,	KC_INSERT,
+		KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_HOME,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_SCROLLLOCK,
+		KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_LEFT,	KC_DOWN,	KC_UP,		KC_RIGHT,	KC_TRNS,	KC_TRNS,				KC_TRNS,	KC_PAUSE,
+		KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_END,	KC_TRNS,	KC_TRNS,	KC_TRNS,		KC_TRNS,				KC_VOLU,	KC_PSCREEN,
+		KC_TRNS,	KC_TRNS,	KC_TRNS,							KC_MPLY,						KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_MPRV,					KC_VOLD,	KC_MNXT
+    ),
+    [_MOUSE_LAYER] = LAYOUT_65_ansi(
+		KC_GRAVE,	KC_F1,				KC_F2,		KC_F3,			KC_F4,		KC_F5,		KC_F6,		KC_F7,		KC_F8,		KC_F9,		KC_F10,		KC_F11,		KC_F12,		KC_TRNS,	KC_TRNS,
+		KC_ACL0,	KC_BTN1,			KC_MS_U,	KC_BTN2,		KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_BTN4,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,
+		KC_ACL1,	KC_MS_L,			KC_MS_D,	KC_MS_R,		KC_TRNS,	KC_TRNS,	KC_WH_L,	KC_WH_D,	KC_WH_U,	KC_WH_R,	KC_TRNS,			KC_TRNS,		KC_TRNS,	KC_TRNS,
+		KC_ACL2,	AUTOCLICK_TOGGLE,	KC_BTN3,	RAPID_CLICK,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_BTN5,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,				KC_TRNS,	KC_TRNS,
+		CYCLE_CLICK_SPEEDS,	KC_TRNS,	KC_TRNS,								KC_TRNS,							KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,				KC_TRNS,	KC_TRNS
+    ),
+    [_HYPER_LAYER] = LAYOUT_65_ansi(
+		DYN_REC_STOP,	DYN_REC_START1,		DYN_REC_START2,		KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,
+		KC_NO,			DYN_MACRO_PLAY1,	DYN_MACRO_PLAY2,	KC_NO,		KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,
+		KC_NO,			KC_NO,				KC_NO,				KC_NO,		KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,				KC_TRNS,	KC_TRNS,
+		KC_LSHIFT,		KC_NO,				KC_NO,				RGB_HUI,	RGB_SAI,	RGB_VAI,	RGB_SPI,	RGB_MOD,	KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,				KC_TRNS,	KC_TRNS,
+		KC_TRNS,		KC_TRNS,			KC_TRNS,									RGB_TOG,							KC_TRNS,	KC_TRNS,	KC_TRNS,	KC_TRNS,				KC_TRNS,	KC_TRNS
+    )
+};
+
+static bool autoclick = false;
+static bool autoclick_state = false;
+static uint16_t autoclick_timer = 0;
+
+int clickspeed_set = 0;
+#define CLICKSPEEDS 4
+const int clickspeed_sets[CLICKSPEEDS] = {0, 20, 100, 1000};
+int press_interval = clickspeed_sets[0];
+
+void matrix_scan_user(void) {
+  if (autoclick && timer_elapsed(autoclick_timer) > (autoclick_state ? 1 : press_interval)) {
+    if (autoclick_state) {
+      mousekey_off(KC_MS_BTN1);
+    } else {
+      mousekey_on(KC_MS_BTN1);
```

Comment:
You could use `tap_code(KC_BTN1)` rather than the mousekey functions.

Reply:
Orignally it had a release timer built in as well. But I scrapped that. Thanks for your help!